### PR TITLE
Adhomian typewriter and camera

### PR DESCRIPTION
**Pull request by Alberyk**
-adds an adhomai typewriter to the loadout
-adds an adhomian camera to the loadout

Sprites made by sue

### CODE REVIEW

**Review by DreamySkrell — APPROVED** [review]

File: code/modules/paperwork/typewriter.dm [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	contained_sprite = TRUE
+	force = 10
+	throwforce = 5
+	throw_speed = 1
+	throw_range = 4
+	w_class = ITEMSIZE_LARGE
+	drop_sound = 'sound/items/drop/backpack.ogg'
+	pickup_sound = 'sound/items/pickup/backpack.ogg'
+
+	var/obj/item/portable_typewriter/machine
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:123]: This reference needs to be removed in Destroy()

File: code/modules/paperwork/typewriter.dm [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	var/opened = FALSE
+
+/obj/item/typewriter_case/New()
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:126]: Should use Initialize over New

File: code/modules/paperwork/typewriter.dm [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	said, rural areas and less urban parts of the planet still rely heavily on these machines."
+	icon_state = "typewriter"
+	icon = 'icons/obj/device.dmi'
+	force = 20
+	throwforce = 5
+	w_class = ITEMSIZE_NORMAL
+	drop_sound = 'sound/items/drop/metalweapon.ogg'
+	pickup_sound = 'sound/items/pickup/metalweapon.ogg'
+
+	var/obj/item/paper/stored_paper = null
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:21]: The reference should be cleared in Destroy()

File: code/modules/paperwork/typewriter.dm [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+	var/obj/item/pen/pen
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:22]: The reference should be cleared in Destroy()

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

File: code/modules/paperwork/typewriter.dm [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+		pen = new /obj/item/pen/typewriter(src)
+
+/obj/item/portable_typewriter/Destroy()
+	if(stored_paper)
+		qdel(stored_paper)
+		stored_paper = null
+
+	if(pen)
+		qdel(pen)
+		pen = null
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:36]:
```suggestion
	QDEL_NULL(stored_paper)
	QDEL_NULL(pen)
```

File: code/modules/paperwork/typewriter.dm [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+/obj/item/typewriter_case/Initialize()
+	. = ..()
+	if(!machine)
+		machine = new /obj/item/portable_typewriter(src)
+
+/obj/item/typewriter_case/Destroy()
+	if(machine)
+		qdel(machine)
+		machine = null
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:145]:
```suggestion
	QDEL_NULL(machine)
```

File: code/modules/paperwork/typewriter.dm [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+		return
+
+	playsound(loc, 'sound/items/storage/briefcase.ogg', 50, 0, -5)
+
+	if(!opened)
+		opened = TRUE
+		update_icon()
+	else
+		opened = FALSE
+		update_icon()
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:159]:
```suggestion
	opened = !opened
	update_icon()
```

File: code/modules/paperwork/typewriter.dm [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+	if(pen)
+		qdel(pen)
+		pen = null
+
+	return ..()
+
+/obj/item/portable_typewriter/attack_self(mob/user)
+	if(!stored_paper)
+		to_chat(user, SPAN_ALERT ("\The [src] has no paper fed for typing!"))
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:42]:
```suggestion
		to_chat(user, SPAN_ALERT("\The [src] has no paper fed for typing!"))
```

File: code/modules/paperwork/typewriter.dm [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+			user.put_in_hands(paper)
+	else
+		. = ..()
+
+/obj/item/portable_typewriter/MouseDrop(mob/user as mob)
+	if(use_check_and_message(user))
+		return
+	if(((user.contents.Find(src) || in_range(src, user))))
+		if(isnull(stored_paper))
+			to_chat(user, SPAN_ALERT ("\The [src] has no paper fed for typing!"))
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:67]:
```suggestion
			to_chat(user, SPAN_ALERT("\The [src] has no paper fed for typing!"))
```

File: code/modules/paperwork/typewriter.dm [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+		return
+	if(((user.contents.Find(src) || in_range(src, user))))
+		if(opened)
+			if(!machine)
+				to_chat(user, SPAN_ALERT("\The [src] is currently empty!"))
+			else
+				user.put_in_hands(machine)
+				machine = null
+				update_icon()
+		else if (!opened)
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:180]:
```suggestion
		else
```

File: code/modules/paperwork/typewriter.dm [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+	attack_self(user)
+
+/obj/item/typewriter_case/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/portable_typewriter))
+		if(!machine)
+			user.drop_item(W)
+			user.unEquip(W)
+			W.forceMove(src)
+			src.machine = W
+			user.visible_message(SPAN_ALERT ("[user] places \the [W] into \the [src]."), SPAN_ALERT ("You store \the [W] in \the [src]."))
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:194]:
```suggestion
			user.visible_message(SPAN_ALERT("[user] places \the [W] into \the [src]."), SPAN_ALERT ("You store \the [W] in \the [src]."))
```

File: code/modules/paperwork/typewriter.dm [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+/obj/item/portable_typewriter/update_icon()
+	if(stored_paper)
+		icon_state = "typewriter_[stored_paper.icon_state]"
+	else
+		icon_state = "typewriter"
+
+/obj/item/portable_typewriter/proc/eject_paper(atom/target, mob/user)
+	if(!stored_paper)
+		return FALSE
+	to_chat(user, SPAN_ALERT ("\The [src] ejects \the [stored_paper]."))
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:81]:
```suggestion

	to_chat(user, SPAN_ALERT ("\The [src] ejects \the [stored_paper]."))
```

File: code/modules/paperwork/typewriter.dm [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+		var/obj/item/paper/paper = stored_paper
+		if(eject_paper(user.loc))
+			user.put_in_hands(paper)
+	else
+		. = ..()
+
+/obj/item/portable_typewriter/MouseDrop(mob/user as mob)
+	if(use_check_and_message(user))
+		return
+	if(((user.contents.Find(src) || in_range(src, user))))
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:65]:
```suggestion

	if(((user.contents.Find(src) || in_range(src, user))))
```

File: code/modules/paperwork/typewriter.dm [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+	w_class = ITEMSIZE_NORMAL
+	drop_sound = 'sound/items/drop/metalweapon.ogg'
+	pickup_sound = 'sound/items/pickup/metalweapon.ogg'
+
+	var/obj/item/paper/stored_paper = null
+	var/obj/item/pen/pen
+
+/obj/item/portable_typewriter/Initialize()
+	. = ..()
+	if(!pen)
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:26]:
```suggestion

	if(!pen)
```

File: code/modules/paperwork/typewriter.dm [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+	if(stored_paper)
+		icon_state = "typewriter_[stored_paper.icon_state]"
+	else
+		icon_state = "typewriter"
+
+/obj/item/portable_typewriter/proc/eject_paper(atom/target, mob/user)
+	if(!stored_paper)
+		return FALSE
+
+	to_chat(user, SPAN_ALERT ("\The [src] ejects \the [stored_paper]."))
```
Comment (FluffyGhoster) [inline code/modules/paperwork/typewriter.dm:79]:
```suggestion
	to_chat(user, SPAN_ALERT("\The [src] ejects \the [stored_paper]."))
```

**Review by FluffyGhoster — APPROVED** [review]
Forgot to remove the space in the previous suggestion, otherwise LGTM

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**